### DISCUSSION TITLE
vcpkg with VS2010/Win7

### DISCUSSION THREAD

**Discussion by nurbles** [microsoft/vcpkg#40381] [Q&A]
I'm getting the dreaded (at least by me) "error: in triplet x64-windows: Unable to find a valid Visual Studio instance" and I don't know how to resolve it, or if it is even _possible_ to resolve.  Here's how I got here:
1. Install vcpkg on my Win7/VS2010 system for the _very first time ever_.  That went fine.
2. Try to run `vcpkg install libzip --triplet=x86-windows` and get the error message above.
3. Notice the **x64** in the message, when I specified **x86** as the triplet, I tried setting the `VCPKG_DEFAULT_TRIPLET` and `VCPKG_DEFAULT_HOST_TRIPLET` environment variables to `x86-windows` and got a different, very unexpected error:  `error: Invalid triplet: x86-windows`
4. Decided to try running within a Visual Studio Command Prompt by using the item from the VS2010 Tools menu to start a shell.
5. Try to install libzip again and get the same errors under the same conditions.

Visual Studio 2010 is installed with all of the default folders, paths, etc. and has been working just fine for 14 years. Apparently, vcpkg has a different view of how things should be installed and doesn't care about how Microsoft installed the product for me.

Hopefully, this wonderful community can help me force vcpkg to use the actual configuration of Visual studio instead of whatever imaginary configuration it is using.

Of, if there is some VS component that I did not originally install, then the error message is junk, since it should tell me _exactly_ what component was not found so that I might at least have a _chance_ to solve this on my own.  If any vcpkg developers are reading this, please take note.

**Comment by Neumann-A** [comment]
VS2010 is not supported for automatic detection and env setup. You'll need some manual triplet setup to make it work. You probably can simply set VCPKG_CHAINLOAD_TOOLCHAIN file to the value vcpkg normally uses. This dectivates the automatic setup but also means you have to run from a dev cmd with everything already set up

**Reply by nurbles** [reply]
Thanks, I'll give that a try.  I found a set of Visual Studio build packages for libzip (and zlib) here on github, but when I call the zip_open() function in it I get an invalid instruction exception.  I'm hoping the vcpkg build will work better, if I can get it to build.  Thanks again.

[EDIT]  Sadly, I clearly do not understand how to follow your suggestion.  I tried using a Visual Studio Command Line (which sets up all of the compiler paths, etc.) and tried setting VCPKG_CHAINLOAD_TOOLCHAIN to the full path to the windows.cmake file (none of the others there seem to fit), then I tried just the file name, then I tried just script/toolchains/windows.cmake, but I get the same "Unable to find" error in all cases.

I also tried with --triplet=x86-windows on the vcpkg command line, with the same result.  When I remove the triplet setting, I also get a message that I "appear to be in a Visual Studip prompt targeting x86 windows but installing for x64".

So the fix is not quite as simple as I hoped.  Any better, more detailed ideas?
